src/Camunda.Worker/Client/FetchAndLockRequest.cs

```diff
@@ -73,6 +73,9 @@ public Topic(string topicName, int lockDuration)
 
             public IReadOnlyCollection<string>? ProcessDefinitionKeyIn { get; set; }
 
+            /// <summary>A dictionary used for filtering tasks based on process instance variable values. The property Key of the dictionary represents a process variable name, while the property value represents the process variable value to filter tasks by.</summary>
+            public IReadOnlyCollection<KeyValuePair<string, string>>? ProcessVariables { get; set; }
```

Comment:
Why do you use `IReadOnlyCollection<KeyValuePair<string, string>>?` and not `IReadOnlyDictionary<string, string>?`

Reply:
No reason in particular other than keeping with the trend of `IReadOnlyCollection`'s in the `FetchAndLockRequest`, and then a more specific implementation in the `HandlerMetadata` class. Like the `IReadOnlyList<string>` or in this case `IReadOnlyDictionary<string, string>`.

And, since one inherits from the other,
`IReadOnlyDictionary<[Nullable(2)] TKey, [Nullable(2)] TValue> : IReadOnlyCollection<KeyValuePair<TKey, TValue>>,`
it should work the same way.

If you prefer it, I can use switch it to `IReadOnlyDictionary<string, string>`. It is really up to your liking.

Reply:
I think that it would be better to use `IReadOnlyDictionary<string, string>` in this case because it is intuitively clear how collection is serialized and we have a guarantee that there will be no two identical keys in the collection